### PR TITLE
SNOW-1653357 Use ApplicationPackageEntity.deploy for both PDF versions

### PR DESCRIPTION
**Pull request by sfc-gh-fcampbell**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Extracts `ApplicationPackageEntity.action_deploy()` into a method that can also be called from `NativeAppManager.deploy()`.

### CODE REVIEW

File: tests/nativeapp/test_manager.py [hunk 1]
```diff
@@ -1701,61 +1701,3 @@ def test_stream_events(mock_execute, mock_account_event_table, temp_dir, mock_cu
...
-
-
-@mock.patch.object(NativeAppManager, "validate")
-@mock.patch.object(NativeAppManager, "execute_package_post_deploy_hooks")
-@mock.patch.object(NativeAppManager, "sync_deploy_root_with_stage")
-@mock.patch.object(NativeAppManager, "_apply_package_scripts")
-@mock.patch.object(NativeAppManager, "create_app_package")
-@mock.patch.object(NativeAppManager, "use_role")
-@mock.patch.object(NativeAppManager, "use_package_warehouse")
-def test_deploy_with_package_post_deploy_hook(
```
Comment (sfc-gh-fcampbell) [inline tests/nativeapp/test_manager.py:1713]: we test deploy in the ApplicationPackageEntity tests, which now covers all PDF versions

File: src/snowflake/cli/_plugins/nativeapp/manager.py [hunk 2]
```diff
@@ -325,35 +325,25 @@ def deploy(
         validate: bool = True,
         print_diff: bool = True,
```
[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/_plugins/nativeapp/manager.py:326]: ah this was `True` unconditionally in v2, i'll add the missing param

File: src/snowflake/cli/api/entities/application_package_entity.py [hunk 3]
```diff
@@ -208,6 +176,94 @@ def bundle(
...
+        package_scripts: List[str],
+    ) -> DiffResult:
+        # 1. Create a bundle
+        bundle_map = cls.bundle(
+            project_root=project_root,
+            deploy_root=deploy_root,
+            bundle_root=bundle_root,
+            generated_root=generated_root,
+            artifacts=artifacts,
+            package_name=package_name,
```
[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/api/entities/application_package_entity.py:207]: can you clarify what you mean? in the previous version we were already using `model.fqn.identifier`

File: src/snowflake/cli/api/entities/application_package_entity.py [hunk 4]
```diff
@@ -208,6 +176,94 @@ def bundle(
...
+                deploy_root=deploy_root,
+                package_name=package_name,
+                stage_schema=stage_schema,
+                bundle_map=bundle_map,
+                role=package_role,
+                prune=prune,
+                recursive=recursive,
+                stage_fqn=stage_fqn,
+                local_paths_to_sync=paths,
+                print_diff=True,
```
[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/api/entities/application_package_entity.py:242]: re-added the flag so `NativeAppManager` can pass it through

File: src/snowflake/cli/_plugins/nativeapp/manager.py [hunk 5]
```diff
@@ -325,35 +325,25 @@ def deploy(
...
-
-            # 4. Execute post-deploy hooks
-            with self.use_package_warehouse():
-                self.execute_package_post_deploy_hooks()
-
-        if validate:
-            self.validate(use_scratch_stage=False)
-
-        return diff
+        return ApplicationPackageEntity.deploy(
```
[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/_plugins/nativeapp/manager.py:328]: yeah that would be idea, but it's not a straightforward 1:1 refactor like this is, so i'd feel more comfortable doing it separately (and it makes review easier)

File: src/snowflake/cli/api/entities/application_package_entity.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sfc-gh-melnacouzi) [reply src/snowflake/cli/api/entities/application_package_entity.py:209]: if you look at self.action_bundle(), which was called from self.action_deploy() before, you can see that it's passing in model.identifier as package_name. See line 73. 
I think it's probably the same value, but please double check. Thanks

**Review by sfc-gh-gbloom — APPROVED** [review]

**Pull request merged**